src/api/app/controllers/webui/webui_controller.rb

```diff
@@ -169,9 +168,36 @@ def authenticate
     mode = CONFIG['proxy_auth_mode'] || :off
     logger.debug "Authenticating with iChain mode: #{mode}"
     if mode == :on || mode == :simulate
-      authenticate_proxy
+      mode = CONFIG['proxy_auth_mode'] || :off
+      proxy_user = request.env['HTTP_X_USERNAME']
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.